model_rnn/adapter.py

```diff
@@ -33,26 +36,16 @@ def _predict_helper(self, title:str, top_n:int=10, max_length:int=100, result_li
                 topv, topi = output.reshape(-1).topk(2)
                 top_char = IX_TO_CHAR[topi[0].item()]
                 top_2_char = IX_TO_CHAR[topi[1].item()]
+
+                if n > 0 and top_2_char != EOS:
+                    n = n // 2
+                    for result in self._predict_helper(title + top_2_char, n, max_length, score + topv[1].item()):
+                        yield result
+
+                score += topv[0].item()
                 if top_char == EOS:
-                    score += topv[0].item()
                     break
-                elif len(result_list) < top_n - 1 and top_2_char != EOS:
-                    for result in self._predict_helper(title + top_2_char, top_n, max_length, result_list, score + topv[1].item()):
-                        if result not in result_list:
-                            #Replace lowest score title with the new result
-                            if result['score'] > min(result_list, key=lambda item: item['score'])['score']:
-                                result_list[result_list.index( min(result_list, key=lambda item: item['score']))] = result
-                            else:
-                                result_list.append(result)
-                score += topv[0].item()
                 title += top_char
                 X = make_input_vect(title)
-            result = {'title':title[1:], 'score':score}
-            if len(result_list) < top_n :
-                result_list.append(result)
-            #Replace lowest score title with the new result
-            elif score > min(result_list, key=lambda item: item['score'])['score']:
-                min_ix = result_list.index( min(result_list, key=lambda item: item['score']))
-                result_list[min_ix] = result
-                
-            return result_list
+
+            yield {'title':title[1:], 'score':score}
```

Comment:
```suggestion
            yield title[1:], score
```
In python it's preferred to return tuples